analyzer/src/main/java/org/jboss/gm/analyzer/alignment/AlignmentTask.java

```diff
@@ -111,54 +107,37 @@ private void writeMarkerFile(File rootDir) throws IOException {
     private Collection<ProjectVersionRef> getAllProjectDependencies(Project project) {
         final Set<ProjectVersionRef> result = new LinkedHashSet<>();
         project.getConfigurations().all(configuration -> {
-            final ResolutionResult resolutionResult = configuration.getIncoming()
-                    .getResolutionResult();//force dependency resolution - this is needed later on if we encounter deps with no version
-            configuration.getAllDependencies().forEach(dep -> {
-                if (dep instanceof DefaultSelfResolvingDependency) {
-                    logger.warn("Ignoring dependency of type {} on project {}", dep.getClass().getName(), project.getName());
-                } else if (isEmpty(dep.getVersion())) {
-                    // look up resolved dependency with a toString hack
-                    boolean found = false;
-                    final Set<? extends DependencyResult> resolvedDependencies = getAllDependenciesFromResolutionResult(
-                            resolutionResult);
-                    for (DependencyResult resolvedDependency : resolvedDependencies) {
-                        if (!(resolvedDependency instanceof DefaultResolvedDependencyResult)) {
-                            continue;
-                        }
 
-                        final String resolvedDependencyDisplayName = ((DefaultResolvedDependencyResult) resolvedDependency)
-                                .getSelected()
-                                .getId().getDisplayName();
-                        // the display name should be something like: "org.example:somelib:1.2.3"
-                        if (resolvedDependencyDisplayName.startsWith(dep.getGroup() + ":" + dep.getName()) && StringUtils
-                                .countMatches(resolvedDependencyDisplayName, ":") == 2) {
-                            final int i = resolvedDependencyDisplayName.lastIndexOf(":");
-                            if (i != -1) {
-                                result.add(ProjectVersionFactory.withGAVAndConfiguration(dep.getGroup(), dep.getName(),
-                                        resolvedDependencyDisplayName.substring(i + 1),
-                                        configuration.getName()));
-                                found = true;
-                            }
+            if (configuration.isCanBeResolved()) {
+
+                LenientConfiguration lenient = configuration.getResolvedConfiguration().getLenientConfiguration();
+
+                if (lenient.getUnresolvedModuleDependencies().size() > 0) {
+                    logger.warn("For configuration {}; unable to resolve all dependencies: {}", configuration,
+                            lenient.getUnresolvedModuleDependencies());
+                }
+
+                lenient.getFirstLevelModuleDependencies().forEach(dep -> {
+                    // TODO: I don't think this is ever possible? If it is then the resolving doesn't work?
```

Comment:
I don't think this is possible now so I'm tempted to remove it.

Reply:
:+1: